pkg/plugin/aggregation/aggregator.go

```diff
@@ -198,21 +212,45 @@ func (a *Aggregator) processResult(result *plugin.Result) error {
 	// Send an event that we got this result even if we get an error, so
 	// that Wait() doesn't hang forever on problems.
 	defer func() {
-		a.Results[result.ExpectedResultID()] = result
+		a.Results[result.Key()] = result
 		a.resultEvents <- result
 	}()
 
 	if err := a.handleResult(result); err != nil {
 		// Drop a breadcrumb so that we reconsider new results from this result.
-		a.FailedResults[result.ExpectedResultID()] = time.Now()
+		a.FailedResults[result.Key()] = time.Now()
 		return &httpError{
 			err:  fmt.Errorf("error handling result %v: %v", resultID, err),
 			code: http.StatusInternalServerError,
 		}
 	}
 
 	// Upon success, we no longer want to keep processing duplicate results.
-	delete(a.FailedResults, result.ExpectedResultID())
+	delete(a.FailedResults, result.Key())
+
+	return nil
+}
+
+// processStatus is the main aggregator logic for handling the status updates from plugins.
+// We first
+func (a *Aggregator) processStatus(status plugin.StatusUpdate) error {
```

Comment:
Add a simple test or two for this; ensuring what/how we check for expected values and handle global nodes, erroring, etc